pubnub-chat-impl/src/commonTest/kotlin/com/pubnub/kmp/ChatTest.kt

```diff
@@ -1655,4 +1702,47 @@ class ChatTest : BaseTest() {
         verify { pubnub.destroy() }
         verify { timerManager.destroy() }
     }
+
+    @Test
+    fun disconnectSubscriptions_shouldCompleteSuccessfully() {
+        // given
+        every { pubnub.disconnect() } returns Unit
+
+        // when
+        objectUnderTest.disconnectSubscriptions().async { result ->
+            // then
+            assertTrue(result.isSuccess)
+        }
+
+        verify { pubnub.disconnect() }
+    }
+
+    @Test
+    fun disconnectSubscriptions_withNoActiveSubscriptions_shouldStillWork() {
+        // given
+        every { pubnub.disconnect() } returns Unit
+        every { pubnub.getSubscribedChannels() } returns emptyList()
+
+        // when - should not throw exception
+        objectUnderTest.disconnectSubscriptions().async { result ->
+            // then
+            assertTrue(result.isSuccess)
+        }
+
+        verify { pubnub.disconnect() }
+    }
+
+    @Test
+    fun reconnectSubscriptions_shouldRestoreConnection() {
```

Comment:
The same as above, what's the difference in a test below and the existing `reconnectSubscriptions_shouldCallPubNubReconnect`?